src/client/modals/druid-query-modal/druid-query-modal.tsx

```diff
@@ -29,8 +30,9 @@ interface DruidQueryModalProps {
 }
 
 export const DruidQueryModal: React.SFC<DruidQueryModalProps> = ({ onClose, timekeeper, essence }) => {
-  const { dataCube: { attributes, source, options: { customAggregations, customTransforms } } } = essence;
-  const query = makeQuery(essence, timekeeper);
+  const { visualization, dataCube: { attributes, source, options: { customAggregations, customTransforms } } } = essence;
+  const queryFn = visualization.name === "grid" ? gridQuery : standardQuery;
```

Comment:
leaking abstraction, visualisation is responsible for "queryfn" so could be move this logic into visualisation itself?

Reply:
Yes, very leaking. These components are very far apart and don't have a good point to communicate. I knowingly made this change in the name of technical debt :)

Reply:
sure, just make a note and we will discuss how to define visualisation-make-query abstraction later on